adafruit_bus_device/i2c_device.py

```diff
@@ -164,8 +167,15 @@ def write_then_readinto(
     # pylint: enable-msg=too-many-arguments
 
     def __enter__(self):
-        while not self.i2c.try_lock():
-            pass
+
+        if self.timeout is not None:
+            lock_start_time = time.monotonic()
+            while not self.i2c.try_lock():
+                if self.timeout > (time.monotonic() - lock_start_time):
+                    raise Exception("'I2CDevice' lock timed out")
```

Comment:
Don't raise bare `Exception`s. It seems other Adafruit libraries use `RuntimeError` here. [source](https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI/blob/master/adafruit_esp32spi/adafruit_esp32spi.py#L205).